### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by giladyosef54**

### CODE REVIEW

File: typescript/usersManager/src/client/client_requests.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export const ensureUserExistence = (username: string, password: string, baseUrl: string = 'http://localhost:3000/',
+        getUserRoute: string = 'getUserDetails/', saveUserRoute: string = 'saveOneUserData') => {
+    const logger = new Logger({
+        minLevel: 3
+    })
+
+    axios.get(baseUrl + getUserRoute + username).then((res: AxiosResponse) => {
+        if (res.data.password == password) logger.info('Passwords match')
+        else console.info('Passwords weren\'t match')
+    }).catch((err) => {
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/client/client_requests.ts]: Logger

Comment (Chavafraenkel) [reply typescript/usersManager/src/client/client_requests.ts]: And change to level error

File: typescript/usersManager/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+app.use('/', router)
+
+
+
+app.listen(port, () => {
+    const logger = new Logger({
+        minLevel: 3
+    })
+    logger.info(`Server running at http://localhost:${port}`);
+    ensureUserExistence('asda', 'sadadsa')
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/index.ts:23]: Move ip to env

File: typescript/usersManager/src/server/crud_username.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+let users: Username[] = [];
+
+
+interface massageResponse {
+    status: number,
+    massage: string
+}
+
+router.post('/saveUsersData', (req: Request, res: Response) => {
+    let status: number = 201
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/server/crud_username.ts:11]: Dont use let

File: typescript/usersManager/src/server/crud_username.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+router.post('/saveOneUserData', (req: Request, res: Response) => {
+    const resParams = tryPushUser(req.body.username, req.body.password)
+    
+    res.status(resParams.status).send(resParams.massage);
+});
+
+
+const tryPushUser = (username: string, password: string): massageResponse => {
+    let status: number
+    let massage: string
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/server/crud_username.ts:38]: Same

File: typescript/usersManager/src/server/crud_username.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    
+    if (users.find((user => user.username = username)))
+    {
+        status = 409
+        massage = 'User already exist, data wasn\'t saved.'
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/server/crud_username.ts:43]: Why 409?

File: typescript/usersManager/src/server/crud_username.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    }
+    else {
+        const newUser: Username = {
+            username: username,
+            password: password
+        };
+        status = 201
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/server/crud_username.ts:50]: Why 201?

File: typescript/usersManager/package-lock.json [hunk 7]
```diff
@@ -0,0 +1,1114 @@
+{
+  "name": "usersmanager",
```
Comment (Chavafraenkel) [inline typescript/usersManager/package-lock.json]: Remove from pr

File: typescript/.gitignore [hunk 8]
```diff
@@ -0,0 +1,2 @@
+/experiments.js
```
Comment (Chavafraenkel) [inline typescript/.gitignore:1]: Why double?

File: typescript/usersManager/src/server/crud_username.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
+import { Router, Request, Response } from 'express';
+import { Username } from '../models/username';
+
+const router = Router();
+let users: Username[] = [];
+
+
+interface massageResponse {
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/server/crud_username.ts]: Move all types to a different file

File: typescript/usersManager/src/client/client_requests.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        if (err.response.status == 404) {
+            logger.info(err.response.data)
+            const newUser: Username = {
+                username: username,
+                password: password
+            }
+            logger.info('Create new user.')
+            axios.post(baseUrl + saveUserRoute, newUser).catch((err) => {
+                if (err.response.status == 409) {
+                    logger.info(err.response.data)
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/client/client_requests.ts]: Same

File: typescript/usersManager/src/client/client_requests.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
+import axios, { AxiosResponse } from 'axios'
+import { Username } from '../models/username'
+import {Logger} from 'tslog'
+
+
+export const ensureUserExistence = (username: string, password: string, baseUrl: string = 'http://localhost:3000/',
+        getUserRoute: string = 'getUserDetails/', saveUserRoute: string = 'saveOneUserData') => {
```
Comment (Chavafraenkel) [inline typescript/usersManager/src/client/client_requests.ts]: Move to env

File: typescript/usersManager/src/server/crud_username.ts [hunk 12]
```diff
@@ -0,0 +1,75 @@
+import { Router, Request, Response } from 'express';
+import { Username, massageResponse } from '../models/types';
+
+const router = Router();
+let users: Username[] = [];
```
Comment (giladyosef54) [inline typescript/usersManager/src/server/crud_username.ts]: const

**Pull request merged**